config/routes.rb

```diff
@@ -1,21 +1,24 @@
 Rails.application.routes.draw do
-
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
   root "welcome#index"
 
-  # resources :dashboard, as: "merchant_dashboard"
-  resources :merchants do
-    member { get "dashboard"}
-    resources :items, controller: "merchant_items", only: [:index]
-    resources :invoices, controller: "merchant_invoices", only: [:index, :show]
+  resources(
+    :merchant,
+    controller: "merchant_site/merchant",
+    only: %i[index show]
+  ) do
+    resources :items, controller: "merchant_site/items", only: [:index]
+    resources :invoices, controller: "merchant_site/invoices", only: [:index]
   end
 
-  scope module: 'admin' do
-    resources :admin, controller: "index", only: [:index]
-  end
+  resources(
+    :admin,
+    controller: "admin_site/admin",
+    only: %i[index]
+  )
 
-  namespace :admin do
+  namespace :admin_site, path: "/admin" do
```

Comment:
the `:admin_site` specifies which folder to look for the controller in, and the `path: "/admin"` specifies the URI that prepends the resources below